test/functional/ssl_profile_creation/test_ssl_profile_creation.py

```diff
@@ -0,0 +1,79 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pprint import pprint as pp
+import sys
+
+from f5.bigip import ManagementRoot
+
+FAKE_CERT = """-----BEGIN CERTIFICATE-----
+MIIBozCCAQwCAQEwDQYJKoZIhvcNAQEFBQAwGjEYMBYGA1UEAxQPY2EtaW50QGFj
+bWUuY29tMB4XDTE2MDUxMTE2MjcyN1oXDTI2MDUwOTE2MjcyN1owGjEYMBYGA1UE
+AxQPc2VydmVyQGFjbWUuY29tMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCn
+48EdxKMaBvd2nBY4Pt3UI9OhWfDt0JHF/FnE0MOR1DYEP9WqlRlDorojVSignlne
+febsSOFxciUGkeYvGTycR58/aAcWfp6lz6gLoTydfi7/XdReQt4JLzH9f0HYdKzz
+z06PjNTOGKtcBipUQjAjtH8HRIfOyatIAiUAHHBrBwIDAQABMA0GCSqGSIb3DQEB
+BQUAA4GBACvhrSKPtZVMvTUz4I0oxpl85IeM6p2X1qNjlSreCtLLp8o55HF1BEdI
+gLdNgzCs1x9/Mi7ZIwA0FySS4s6E8hMA3OG5Z/42aKGCJyHybeoWXsiVYIwkQi/J
+7293ACoLewtjwaGAFeSijukgyHooJUkqWi/oG8qc+K2GwZ8yeYYi
+-----END CERTIFICATE-----"""
+
+FAKE_KEY = """-----BEGIN RSA PRIVATE KEY-----
+MIICXQIBAAKBgQCn48EdxKMaBvd2nBY4Pt3UI9OhWfDt0JHF/FnE0MOR1DYEP9Wq
+lRlDorojVSignlnefebsSOFxciUGkeYvGTycR58/aAcWfp6lz6gLoTydfi7/XdRe
+Qt4JLzH9f0HYdKzzz06PjNTOGKtcBipUQjAjtH8HRIfOyatIAiUAHHBrBwIDAQAB
+AoGBAIKTPZBEbkIA5xhlv1ZRdr/WeXNFe3/KtoWQhdTwNRrHPJfDeg+o1LRo7HIs
+emOppOXJb/+Xk1djWm6orKk27I6wgIGcLJv61jRq9mOG3Hlfs9ZSkVsQIqutUSVm
+amhp3uwK3KIk3k7yv16+VTGfsXsPWsT1oWd4CWmWNAjMol9JAkEA3IloSbt+orEf
+x36qv//jsq87gOr5eUmwXTySMHaxbXmkIjaCjZHAb70/kWewXZMoj7k6c1Pj9i3T
+Tdfhgl3eLQJBAMLjFS1xIqXKLeBzMjcBfVlDF/ZJa4e2EZd1OOJreGkllx3j23fU
+NBdsN71XGr16RuxkLo/4HozequTCh5fU4oMCQAfFG5SFc5e9z9XSg6eSF26jN+B5
+5uI8E2eli60DcYre30aJTx43xWTqcQPpeFBDsAkoSIPpr71rrecvNPXH4t0CQGGB
+JbZPlUsnZV6Xo/b7StCfDd0ODLugbxq87lHx/RN2WC3/M223gLx7S0Py0ZEdHWDm
+GpmzRO2r9gpv/VEMlKsCQQCV+EffCQ4wKBIYeCchdntop1/A9PWWCS+pjUNdIJNR
+CKxlxUfEZw9yNfLw9g0FKxrdSZiHCAw7fwN7s+CszjT4
+-----END RSA PRIVATE KEY-----"""
+
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+logger.addHandler(logging.StreamHandler(sys.stdout))
+
+
+def test_ssl_profile_creation(symbols, tmpdir):
+    testdatadir = tmpdir.mkdir('testdir')
+    CERTFILENAME = 'fakecert.crt'
+    certpath = testdatadir.join(CERTFILENAME)
+    certpath.write(FAKE_CERT)
+    keypath = testdatadir.join('fakekey')
+    keypath.write(FAKE_KEY)
+    mr = ManagementRoot(symbols.bigip_netloc,
+                        symbols.bigip_username,
+                        symbols.bigip_password)
+    # uploader = mr.shared.file_transfer.uploads
+    cert_registrar = mr.tm.sys.crypto.certs
+    pp(cert_registrar.raw)
+    # key_registrar = mr.tm.sys.crypto.keys
+    ssl_client_profile = mr.tm.ltm.profile.client_ssls.client_ssl
+    # uploader.upload_file(str(certpath))
+    # uploader.upload_file(str(keypath))
+    # cert_registrar.install_cert(CERTFILENAME)
+    # pp(cert_registrar.raw)
+    # key_registrar.install_key(CERTFILENAME, 'fakekey')
+    # pp(ssl_client_profile.get_collection())
+    # ssl_client_profile.build_profile(CERTFILENAME, 'fakekey')
+    # profile1 = ssl_client_profile.create(certname='fakecert.crt',
```

Comment:
@caphrim007   This is where I get the:

"`Client SSL profile must have RSA certificate/key pair.`"

Error.